integration/src/main/java/com/aws/greengrass/mqtt/moquette/ClientDeviceAuthorizer.java

```diff
@@ -112,12 +116,7 @@ public boolean checkValid(String clientId, String username, byte[] password) {
     @Override
     public boolean canWrite(Topic topic, String user, String client) {
         String resource = "mqtt:topic:" + topic;
-        boolean canPerform = false;
-        try {
-            canPerform = canDevicePerform(getOrCreateSessionForClient(client, user), "mqtt:publish", resource);
-        } catch (AuthenticationException e) {
-            LOG.atWarn().cause(e).kv(CLIENT_ID, client).log("Unable to re-authenticate client.");
-        }
```

Comment:
how come there's no try-catch here anymore?

Reply:
The try-catch was for `getOrCreateSessionForClient` that we passed as a parameter into `canDevicePerform`. In my changes, I pushed the `getOrCreateSessionForClient` call into the `canDevicePerform` function and brought the try-catch block with it to catch if `createdSessionForClient` throws an Auth exception